[run-webkit-tests] Add -2 option

JonWBedard opened the pull request:
#### 9f04f384e3f352b1156bec82cc3b14fb9d76f716
<pre>
[run-webkit-tests] Add -2 option
<a href="https://rdar.apple.com/165717647">rdar://165717647</a>
<a href="https://bugs.webkit.org/show_bug.cgi?id=303418">https://bugs.webkit.org/show_bug.cgi?id=303418</a>

Reviewed by Sam Sneddon.

Add a -2 option which forces ports to run with modern WebKit. Lay down groundwork
for running with multiple drivers. Replace webkit_test_runner with the driver_name variable,
and refactor various checks into parent port classes.

* Tools/Scripts/webkitpy/common/test_expectations_unittest.py:
* Tools/Scripts/webkitpy/layout_tests/controllers/manager.py:
(Manager.run): Check driver_name to determine if we&apos;re running under WebKitLegacy.
* Tools/Scripts/webkitpy/layout_tests/run_webkit_tests.py:
(parse_args): Add -2 option. Check that the user did not specify multiple drivers.
(_set_up_derived_options): Remove special case for GTK and WPE.
* Tools/Scripts/webkitpy/layout_tests/run_webkit_tests_integrationtest.py: Rebaseline.
* Tools/Scripts/webkitpy/layout_tests/views/printing_unittest.py: Ditto.
* Tools/Scripts/webkitpy/performance_tests/perftestsrunner.py:
(PerfTestsRunner.__init__): Remove special case for GTK and WPE.
(PerfTestsRunner._parse_args): Add -2 option. Check that the user did not specify multiple drivers.
* Tools/Scripts/webkitpy/port/apple.py:
(ApplePort.determine_full_port_name): Use driver check.
* Tools/Scripts/webkitpy/port/base.py:
(Port): Set driver names as a constant.
(Port.is_webkitlegacy): Check if the specified driver is a WebKitLegacy driver.
(Port.determine_full_port_name): Use driver check.
(Port.determine_driver_name): Centralize options to driver name conversions.
(Port.__init__): Check that the specified driver is supported for this port.
(Port.default_baseline_search_path): Use a driver check.
(Port.driver_name): Standardize driver check.
(Port._port_specific_expectations_files): Use a driver check.
(Port._build_driver): Ditto.
* Tools/Scripts/webkitpy/port/darwin.py:
(DarwinPort.__init__): Use modern `super()` call.
* Tools/Scripts/webkitpy/port/darwin_testcase.py:
(DarwinTest.assert_skipped_file_search_paths): Remove unused function.
* Tools/Scripts/webkitpy/port/device_port.py:
(DevicePort.driver_name): Convert driver name for Apple embedded platforms.
* Tools/Scripts/webkitpy/port/driver_unittest.py:
(DriverTest.test_no_timeout): Rebaseline.
* Tools/Scripts/webkitpy/port/factory.py:
(PortFactory.get): Never vend invalid ports.
* Tools/Scripts/webkitpy/port/ios.py:
(IOSPort): Support both WebKitTestRunner and DumpRenderTree.
(IOSPort.default_baseline_search_path): Use a driver check.
* Tools/Scripts/webkitpy/port/ios_device_unittest.py: Rebaseline.
* Tools/Scripts/webkitpy/port/ios_simulator_unittest.py: Rebaseline.
* Tools/Scripts/webkitpy/port/ios_testcase.py: Rebaseline.
* Tools/Scripts/webkitpy/port/mac.py:
(MacPort): Support both WebKitTestRunner and DumpRenderTree.
(MacPort.__init__): Use modern `super()` call.
(MacPort.default_baseline_search_path): Use a driver check.
(MacPort.default_child_processes): Remove special cases for WebKit.
* Tools/Scripts/webkitpy/port/mac_unittest.py: Rebaseline.
* Tools/Scripts/webkitpy/port/port_testcase.py: Rebaeline.
* Tools/Scripts/webkitpy/port/test.py: Support both WebKitTestRunner and DumpRenderTree.
* Tools/Scripts/webkitpy/port/visionos.py:
(VisionOSPort.__init__): Deleted.
(VisionOSPort.driver_name): Deleted.
* Tools/Scripts/webkitpy/port/visionos_testcase.py:
(VisionOSTest.make_port): Remove special case.
* Tools/Scripts/webkitpy/port/watch.py:
(WatchPort.__init__): Deleted.
(WatchPort.driver_name): Deleted.
* Tools/Scripts/webkitpy/port/watch_testcase.py:
(WatchTest.make_port): Remove special case.
* Tools/Scripts/webkitpy/port/win.py:
(WinPort): Support both WebKitTestRunner and DumpRenderTree.
(WinPort.do_text_results_differ): Use a driver check.
(WinPort.driver_name): Convert driver name for Windows.
(WinPort._search_paths): Use a driver check.
* Tools/Scripts/webkitpy/port/win_unittest.py:
(WinPortTest._assert_search_path): Set driver.
(WinPortTest.test_compare_text): Rebaseline.
(WinPortTest.test_expectations_files): Rebaseline.
* Tools/Scripts/webkitpy/style/checkers/test_expectations.py:
(TestExpectationsChecker._determine_port_from_expectations_path): Handle ports which
do not support WebKitLegacy.

Canonical link: <a href="https://commits.webkit.org/304573@main">https://commits.webkit.org/304573@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/d203d4065fb840a6d7e0bec6a7c66fd02171f241

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/135963 "3 style errors") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/8325 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/47245 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/143670 "Built successfully") | [  ~~🛠 win~~](https://ews-build.webkit.org/#/builders/59/builds/88177 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/a4755f23-c156-417b-8dcf-52b616eb9a83) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/137832 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/8990 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/8171 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/103905 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/59/builds/88177 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/1c9d053a-e6a3-448d-8c3b-6f2afcac5332) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/138909 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/162/builds/6506 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/121847 "Passed tests") | [❌ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/84782 "Found 1 new API test failure: WPE/TestWebKitWebView:/webkit/WebKitWebView/save (failure)") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/c7d7ad29-e526-4b77-9128-3665be5903f8) 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/135311 "Passed tests") | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/154/builds/6200 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 api-mac-debug](https://ews-build.webkit.org/#/builders/165/builds/3850 "Passed tests") | [✅ 🛠 wpe-cairo-libwebrtc](https://ews-build.webkit.org/#/builders/166/builds/4273 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/115466 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/40038 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/146421 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/8009 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/40606 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/112256 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/8029 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/161/builds/6722 "Passed tests") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/112649 "Found 36 new API test failures: WebKitGTK/TestWebKitAccessibility:/webkit/WebKitAccessibility/text/basic, WebKitGTK/TestWebKitAccessibility:/webkit/WebKitAccessibility/document/basic, WebKitGTK/TestWebKitAccessibility:/webkit/WebKitAccessibility/accessible/ignored-objects, WebKitGTK/TestWebKitAccessibility:/webkit/WebKitAccessibility/hyperlink/basic, WebKitGTK/TestWebKitAccessibility:/webkit/WebKitAccessibility/text/extents, WebKitGTK/TestWebKitAccessibility:/webkit/WebKitAccessibility/selection/menulist, WebKitGTK/TestWebKitAccessibility:/webkit/WebKitAccessibility/accessible/basic-hierarchy, WebKitGTK/TestWebKitAccessibility:/webkit/WebKitAccessibility/action/basic, WebKitGTK/TestWebKitAccessibility:/webkit/WebKitAccessibility/accessible/state-changed/focus, WebKitGTK/TestWebKitAccessibility:/webkit/WebKitAccessibility/image/basic ... (failure)") | | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/6110 "Passed tests") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/118148 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/61909 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/20945 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/8057 "Built successfully") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/36220 "Passed tests") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/7779 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/8000 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/7860 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

stwrt reviewed (commented):
Could you add a brief comment in the commit message on what the `-2` flag does.

aproskuryakov commented on Tools/Scripts/webkitpy/layout_tests/run_webkit_tests.py:
```diff
@@ -145,6 +145,8 @@ def parse_args(args):
             help="Run a concurrent JavaScript thread with each test"),
         optparse.make_option("--dump-render-tree", "-1", action="store_false", default=True, dest="webkit_test_runner",
             help="Use DumpRenderTree rather than WebKitTestRunner. This runs the wk1 single-process architecture."),
+        optparse.make_option("--webkit-test-runner", "-2", action="store_true", default=True, dest="webkit_test_runner",
+            help="Use WebKitTestRunner exclusively, skip any wk1 specific tests."),
```
How does it work for both to write into the same variable, and to have a default value? Will default value for `--webkit-test-runner` override the explicit option for `--dump-render-tree`?

JonWBedard replied:
That requires adopting `argparse` instead of `optparse`

aproskuryakov replied:
Or use two variables for the parser, plus a little logic to reconcile them after the fact.

JonWBedard replied:
I did a different version where we ensure that we don't have both options specified. I'd like to switch everything to `argparse` at some point, but that turns out to be quite a lot of work, when I briefly looked into it.

aproskuryakov replied:
I think that this different version is fragile and unnecessarily convoluted, as it duplicates pasting behavior. Was there some obstacle when doing it in the way that I suggested?

JonWBedard replied:
I was trying to avoid using two variables, that's going to end up being a larger change, things we're sort of planning on doing eventually anyways, but my hope was that this would be a quick change...apparently not

aproskuryakov commented on Tools/Scripts/webkitpy/layout_tests/run_webkit_tests_integrationtest.py:
```diff
@@ -732,6 +732,18 @@ def test_crash_log(self):
 
+    def test_crash_log_webkit_test_runner(self):
+        # FIXME: Need to rewrite these tests to not be mac-specific, or move them elsewhere.
+        # Currently CrashLog uploading only works on Darwin and Windows.
```
Not sure why this has to test through crash log path, this seems worth explaining in ChangeLog.

JonWBedard replied:
This is the only test that tests `-1`, I wanted to add one that tests `-2` also.

gsnedders replied:
I don't see where `-1` is tested?

JonWBedard replied:
Right above in `def test_crash_log(self):`, it's actually the more verbose flag, `--dump-render-tree`, that gets tested.

gsnedders replied:
Oh, doh. I just searching for `-1` and didn't look for the longer form.

gsnedders commented on Tools/Scripts/webkitpy/port/base.py:
```diff
@@ -109,13 +110,18 @@ class Port(object):
     _websocket_secure_server = None
     _websocket_server = None
 
+    @classmethod
+    def is_webkitlegacy(cls, driver):
+        return 'dumprendertree' in driver.lower()
```
We very often are doing `self.is_webkitlegacy(self.driver_name())`, which feels a bit silly.

I might be tempted to have two methods: a class method like `is_webkitlegacy_driver(cls, driver)` and an instance method like `is_webkitlegacy(self)`?

gsnedders commented on Tools/Scripts/webkitpy/performance_tests/perftestsrunner.py:
```diff
@@ -122,8 +118,10 @@ def _expand_path(option, opt_str, value, parser):
                 help="Upload the generated JSON file to the specified server when --output-json-path is present."),
-            optparse.make_option("--dump-render-tree", "-1", action="store_false", default=True, dest="webkit_test_runner",
-                help="Use DumpRenderTree rather than WebKitTestRunner."),
+            optparse.make_option("--dump-render-tree", "-1", action='append_const', dest='driver_names', const="DumpRenderTree", default=[],
```
Nit: `default=[]` is redundant here (it's the default default with `append_const`).

JonWBedard replied:
It's not redundant in `optparse` with however our tests work, sadly.

gsnedders replied:
Huh, [the 3.9 docs claim it _should_ work](https://docs.python.org/3.9/library/optparse.html#:~:text=If%20no%20default%20value%20for%20dest%20is%20supplied%2C%20an%20empty%20list%20is%20automatically%20created%20when%20optparse%20first%20encounters%20this%20option%20on%20the%20command%2Dline.).

gsnedders commented on Tools/Scripts/webkitpy/performance_tests/perftestsrunner.py:
```diff
@@ -122,8 +118,10 @@ def _expand_path(option, opt_str, value, parser):
-                help="Use DumpRenderTree rather than WebKitTestRunner."),
+            optparse.make_option("--dump-render-tree", "-1", action='append_const', dest='driver_names', const="DumpRenderTree", default=[],
+                help="Use DumpRenderTree rather than WebKitTestRunner. This runs the wk1 single-process architecture."),
+            optparse.make_option("--webkit-test-runner", "-2", action='append_const', dest='driver_names', const="WebKitTestRunner", default=[],
```
Nit: `default=[]` is redundant here (it's the default default with `append_const`).

gsnedders commented on Tools/Scripts/webkitpy/layout_tests/run_webkit_tests.py:
```diff
@@ -143,8 +143,10 @@ def parse_args(args):
         optparse.make_option("--threaded", action="store_true", default=False,
             help="Run a concurrent JavaScript thread with each test"),
-        optparse.make_option("--dump-render-tree", "-1", action="store_false", default=True, dest="webkit_test_runner",
+        optparse.make_option("--dump-render-tree", "-1", action='append_const', dest='driver_names', const="DumpRenderTree", default=[],
```
Nit: `default=[]` is redundant here (it's the default default with `append_const`).

gsnedders commented on Tools/Scripts/webkitpy/layout_tests/run_webkit_tests.py:
```diff
@@ -143,8 +143,10 @@ def parse_args(args):
-        optparse.make_option("--dump-render-tree", "-1", action="store_false", default=True, dest="webkit_test_runner",
+        optparse.make_option("--dump-render-tree", "-1", action='append_const', dest='driver_names', const="DumpRenderTree", default=[],
             help="Use DumpRenderTree rather than WebKitTestRunner. This runs the wk1 single-process architecture."),
+        optparse.make_option("--webkit-test-runner", "-2", action='append_const', dest='driver_names', const="WebKitTestRunner", default=[],
```
Nit: `default=[]` is redundant here (it's the default default with `append_const`).

gsnedders commented on Tools/Scripts/webkitpy/port/device_port.py:
```diff
@@ -69,11 +69,8 @@ def child_processes(self):
-            return 'WebKitTestRunnerApp.app'
-        return 'DumpRenderTree.app'
+        parent = super(DevicePort, self).driver_name()
+        return dict(WebKitTestRunner='WebKitTestRunnerApp.app').get(parent, f'{parent}.app')
```
This is the same as `f'{parent}.app`, given the only special case (via a dict? why?) does exactly the same thing.

JonWBedard replied:
Not exactly the same, frustratingly, `WebKitTestRunnerApp.app` vs `WebKitTestRunner.app`.

gsnedders replied:
I still don't see the reason for the `dict` — constructing a `dict`, looking up with a hash, versus and `if` statement and comparing to a (interned) string doesn't seem worthwhile?

gsnedders replied:
Also ping for this thread — I'd still recommend using an `if` statement rather than this `dict` usage.

gsnedders commented on Tools/Scripts/webkitpy/port/factory.py:
```diff
@@ -150,7 +150,10 @@ def get(self, port_name=None, options=None, **kwargs):
-                return cls(self._host, port_name, options=options, **kwargs)
+                try:
+                    return cls(self._host, port_name, options=options, **kwargs)
+                except ValueError:
```
When do we expect this to raise? Do we want to log anything when this happens?

JonWBedard replied:
We don't want to log anything, the exception being raised on line `157` is the correct error to raise, in this case. We expect this to happen when someone requests an invalid port, like `visionos-wk1`, for example.

gsnedders replied:
Ah, and we want to keep trying the rest of the classes in case they match, to raise the other exception at the end?

gsnedders commented on Tools/Scripts/webkitpy/port/win.py:
```diff
@@ -59,12 +59,10 @@ class WinPort(ApplePort):
     DEFAULT_ARCHITECTURE = 'x86_64'
-
     CRASH_LOG_PREFIX = "CrashLog"
+    DRIVER_NAMES = ('WebKitTestRunner', 'DumpRenderTree')
```
Nit (and not really something I expect addressed here): does any current port on Windows support WK1? I thought with the death of Apple's Windows WebKit port that was gone?

JonWBedard replied:
I don't believe they do, and I _think_ the entirety of the `win.py` file can be removed, but I didn't want to chase down that dragon right this moment.

gsnedders replied:
Ah, no, it is used by `wincairo`; c.f. https://github.com/WebKit/WebKit/commit/099cdafc1313f7d09867dc8c86ee1fde67c4fdaa and https://github.com/WebKit/WebKit/commit/c8fc8c9e214ac645769900a3f1315b3c9976cd34.

Though I don't believe they support WK1.

gsnedders commented on Tools/Scripts/webkitpy/port/base.py:
```diff
@@ -109,13 +110,18 @@ class Port(object):
         options = options or {}
         assert port_name.startswith(cls.port_name)
-        if getattr(options, 'webkit_test_runner', False) and not '-wk2' in port_name:
+        driver = getattr(options, 'driver_name', None) or (getattr(options, 'driver_names', []) or cls.DRIVER_NAMES)[0]
```
What's the reason to support both `options.driver_name` _and_ `options.driver_names`?

gsnedders commented on Tools/Scripts/webkitpy/port/base.py:
```diff
@@ -109,13 +110,18 @@ class Port(object):
         options = options or {}
         assert port_name.startswith(cls.port_name)
-        if getattr(options, 'webkit_test_runner', False) and not '-wk2' in port_name:
+        driver = (getattr(options, 'driver_names', []) or cls.DRIVER_NAMES)[0]
```
We have this duplicated a few places, and it's very similar to `Port.driver_name(self)`.

I sure wouldn't mind having a `Port.determine_driver_name(cls, options)` which does this, and then `Port.driver_name(self)` can just call that with `self._options`.

gsnedders commented on Tools/Scripts/webkitpy/port/base.py:
```diff
@@ -134,7 +140,10 @@ def __init__(self, host, port_name, options=None, **kwargs):
 
         if self._name and '-wk2' in self._name:
-            self._options.webkit_test_runner = True
+            self._options.driver_name = self.DRIVER_NAMES[0]
```
How come we aren't looking at `options.driver_names` here?

JonWBedard replied:
We should be, this got update right after your feedback.

gsnedders commented on Tools/Scripts/webkitpy/port/base.py:
```diff
@@ -134,7 +140,10 @@ def __init__(self, host, port_name, options=None, **kwargs):
-            self._options.webkit_test_runner = True
+            self._options.driver_name = self.DRIVER_NAMES[0]
+
+        if not any([self.driver_name().startswith(candidate) for candidate in self.DRIVER_NAMES]):
```
`str.startswith` accepts a tuple directly:

```suggestion
        if not self.driver_name().startswith(self.DRIVER_NAMES):
```

gsnedders commented on Tools/Scripts/webkitpy/port/base.py:
```diff
@@ -134,7 +143,10 @@ def __init__(self, host, port_name, options=None, **kwargs):
+            self._options.driver_names = [self.DRIVER_NAMES[0]]
+
+        if not self.driver_name().startswith(self.DRIVER_NAMES):
+            raise ValueError(f'{self.driver_name()} is not supported on this platform.')
```
Nit: I'd probably be tempted to raise a custom exception type here, something like `UnsupportedDriverError`.

JonWBedard committed:
[run-webkit-tests] Add -2 option
rdar://165717647
https://bugs.webkit.org/show_bug.cgi?id=303418

Reviewed by Sam Sneddon.

Add a -2 option which forces ports to run with modern WebKit. Lay down groundwork
for running with multiple drivers. Replace webkit_test_runner with the driver_name variable,
and refactor various checks into parent port classes.

* Tools/Scripts/webkitpy/common/test_expectations_unittest.py:
* Tools/Scripts/webkitpy/layout_tests/controllers/manager.py:
(Manager.run): Check driver_name to determine if we're running under WebKitLegacy.
* Tools/Scripts/webkitpy/layout_tests/run_webkit_tests.py:
(parse_args): Add -2 option. Check that the user did not specify multiple drivers.
(_set_up_derived_options): Remove special case for GTK and WPE.
* Tools/Scripts/webkitpy/layout_tests/run_webkit_tests_integrationtest.py: Rebaseline.
* Tools/Scripts/webkitpy/layout_tests/views/printing_unittest.py: Ditto.
* Tools/Scripts/webkitpy/performance_tests/perftestsrunner.py:
(PerfTestsRunner.__init__): Remove special case for GTK and WPE.
(PerfTestsRunner._parse_args): Add -2 option. Check that the user did not specify multiple drivers.
* Tools/Scripts/webkitpy/port/apple.py:
(ApplePort.determine_full_port_name): Use driver check.
* Tools/Scripts/webkitpy/port/base.py:
(Port): Set driver names as a constant.
(Port.is_webkitlegacy): Check if the specified driver is a WebKitLegacy driver.
(Port.determine_full_port_name): Use driver check.
(Port.determine_driver_name): Centralize options to driver name conversions.
(Port.__init__): Check that the specified driver is supported for this port.
(Port.default_baseline_search_path): Use a driver check.
(Port.driver_name): Standardize driver check.
(Port._port_specific_expectations_files): Use a driver check.
(Port._build_driver): Ditto.
* Tools/Scripts/webkitpy/port/darwin.py:
(DarwinPort.__init__): Use modern `super()` call.
* Tools/Scripts/webkitpy/port/darwin_testcase.py:
(DarwinTest.assert_skipped_file_search_paths): Remove unused function.
* Tools/Scripts/webkitpy/port/device_port.py:
(DevicePort.driver_name): Convert driver name for Apple embedded platforms.
* Tools/Scripts/webkitpy/port/driver_unittest.py:
(DriverTest.test_no_timeout): Rebaseline.
* Tools/Scripts/webkitpy/port/factory.py:
(PortFactory.get): Never vend invalid ports.
* Tools/Scripts/webkitpy/port/ios.py:
(IOSPort): Support both WebKitTestRunner and DumpRenderTree.
(IOSPort.default_baseline_search_path): Use a driver check.
* Tools/Scripts/webkitpy/port/ios_device_unittest.py: Rebaseline.
* Tools/Scripts/webkitpy/port/ios_simulator_unittest.py: Rebaseline.
* Tools/Scripts/webkitpy/port/ios_testcase.py: Rebaseline.
* Tools/Scripts/webkitpy/port/mac.py:
(MacPort): Support both WebKitTestRunner and DumpRenderTree.
(MacPort.__init__): Use modern `super()` call.
(MacPort.default_baseline_search_path): Use a driver check.
(MacPort.default_child_processes): Remove special cases for WebKit.
* Tools/Scripts/webkitpy/port/mac_unittest.py: Rebaseline.
* Tools/Scripts/webkitpy/port/port_testcase.py: Rebaeline.
* Tools/Scripts/webkitpy/port/test.py: Support both WebKitTestRunner and DumpRenderTree.
* Tools/Scripts/webkitpy/port/visionos.py:
(VisionOSPort.__init__): Deleted.
(VisionOSPort.driver_name): Deleted.
* Tools/Scripts/webkitpy/port/visionos_testcase.py:
(VisionOSTest.make_port): Remove special case.
* Tools/Scripts/webkitpy/port/watch.py:
(WatchPort.__init__): Deleted.
(WatchPort.driver_name): Deleted.
* Tools/Scripts/webkitpy/port/watch_testcase.py:
(WatchTest.make_port): Remove special case.
* Tools/Scripts/webkitpy/port/win.py:
(WinPort): Support both WebKitTestRunner and DumpRenderTree.
(WinPort.do_text_results_differ): Use a driver check.
(WinPort.driver_name): Convert driver name for Windows.
(WinPort._search_paths): Use a driver check.
* Tools/Scripts/webkitpy/port/win_unittest.py:
(WinPortTest._assert_search_path): Set driver.
(WinPortTest.test_compare_text): Rebaseline.
(WinPortTest.test_expectations_files): Rebaseline.
* Tools/Scripts/webkitpy/style/checkers/test_expectations.py:
(TestExpectationsChecker._determine_port_from_expectations_path): Handle ports which
do not support WebKitLegacy.

Canonical link: https://commits.webkit.org/304573@main
